### ISSUE TITLE
RobustMultiService

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11320]
|<img alt="tjs's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| tjs reported|
|-|-|
|Trac ID|trac#496|
|Type|enhancement|
|Created|2004-02-10 12:13:22Z|

Attachments:

* [RobustMultiService.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/93e/93e097e319c2db13edf06f45c4dff4f741394c0d/20936f713cb40d222eb9591b0d61c8fc9f7e59c2.patch) (2103 bytes) - added by tjs on 2004-02-10 12:13:22Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__496 496
type__enhancement enhancement
reporter__tjs tjs
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1076415202000000 1076415202000000
changetime__1471560802042098 1471560802042098
version__ 
owner__ 
cc__moshez cc__tjs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="tjs's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|tjs commented|
|-|-|

```
#!html
<pre>
Here is a new RobustMultiService I wrote to get around the issue of a particular
service raising an exception causes subsequent services not to start. I ran into
this problem when UNIXService finds a dead socket file. I tested this with my
application, hopfully its useful to others.

-tjs
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Duplicate of 495
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

In changeset 36adf49661456829c085e5e35fc4b0fe3222d3ae
```
#!CommitTicketReference repository="" revision="36adf49661456829c085e5e35fc4b0fe3222d3ae"
Merge pull request #496 from twisted/8757-rodrigc-compat-py3

Author: hawkowl, rodrigc
Reviewer: acabhishek942
Fixes: #8757

Fix test_compat on Windows with Python 3
```
